spring-apps/autosleep-core/src/main/java/org/cloudfoundry/autosleep/ui/web/controller/AutoEnrollmentController.java

```diff
@@ -0,0 +1,143 @@
+package org.cloudfoundry.autosleep.ui.web.controller;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import javax.servlet.http.HttpServletResponse;
+
+import org.cloudfoundry.autosleep.access.cloudfoundry.CloudFoundryApiService;
+import org.cloudfoundry.autosleep.access.cloudfoundry.CloudFoundryException;
+import org.cloudfoundry.autosleep.access.dao.model.OrgEnrollmentConfig;
+import org.cloudfoundry.autosleep.access.dao.repositories.OrgEnrollmentConfigRepository;
+import org.cloudfoundry.autosleep.config.Config;
+import org.cloudfoundry.autosleep.config.EnrollmentConfig;
+import org.cloudfoundry.autosleep.ui.servicebroker.service.parameters.ParameterReader;
+import org.cloudfoundry.autosleep.ui.web.controller.validation.EnrollmentValidator;
+import org.cloudfoundry.autosleep.ui.web.model.OrgEnrollmentConfigRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindException;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@RequestMapping(EnrollmentConfig.Path.ORG_AUTO_ENROLMENT_BASE_PATH)
+@RestController
+public class AutoEnrollmentController {
+
+    @Autowired
+    private OrgEnrollmentConfigRepository orgEnrollmentRepository;
+
+    @Autowired
+    private EnrollmentValidator validator;
+
+    @Autowired
+    private CloudFoundryApiService cloudfoundryApi;
+
+    @Autowired
+    private ParameterReader<Config.ServiceInstanceParameters.Enrollment> autoEnrollmentReader;
+
+    @Autowired
+    private ParameterReader<Pattern> excludeFromAutoEnrollmentReader;
+
+    @Autowired
+    private ParameterReader<Duration> idleDurationReader;
+
+    @Autowired
+    private ParameterReader<EnrollmentConfig.EnrollmentParameters.EnrollmentState> stateReader;
```

Comment:
Didn’t get the point. We are using the stateReader and the other readers to build the OrgEnrollmentConfig object from the request data. The default value will remain "back office_enrolled" that the poller will use in case not specified via the REST APIs.